src/Land/Model/RadiativeEnergyFlux.jl

```diff
@@ -0,0 +1,111 @@
+module RadiativeEnergyFlux
+
+using ...VariableTemplates
+using DocStringExtensions
+
+export AbstractNetSwFluxModel,
+    PrescribedSwFluxAndAlbedo,
+    PrescribedNetSwFlux,
+    compute_net_sw_flux,
+    compute_net_radiative_energy_flux
+
+"""
+   AbstractNetSwFluxModel{FT <: AbstractFloat}
+"""
+abstract type AbstractNetSwFluxModel{FT <: AbstractFloat} end
+
+"""
+    PrescribedSwFluxAndAlbedo{FT, FN1, FN2} <: AbstractNetSwFluxModel{FT}
+
+Structure which contains functions for shortwave albedo and
+shortwave fluxes. They are user-defined, constant across the domain
+and can be a function of time.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedSwFluxAndAlbedo{FT, FN1, FN2} <: AbstractNetSwFluxModel{FT}
+    "Shortwave albedo in grid. No units."
+    α::FN1
+    "Shortwave flux in grid. Units of J m-2 s-1"
+    swf::FN2
+    function PrescribedSwFluxAndAlbedo(
+        ::Type{FT};
+        α::FN1 = (t) -> FT(NaN),
+        swf::FN2 = (t) -> FT(NaN),
+    ) where {FT, FN1, FN2}
+        args = (α, swf)
+        return new{FT, FN1, FN2}(args...)
+    end
+end
+
+"""
+    PrescribedNetSwFlux{FT} <: AbstractNetSwFluxModel{FT}
+
+Structure which contains net shortwave flux values. The function is
+user-defined, constant across the domain and can be a function of time.
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct PrescribedNetSwFlux{FT, FN3} <: AbstractNetSwFluxModel{FT}
+    "Net shortwave flux. Units of J m-2 s-1"
+    nswf::FN3
+    function PrescribedNetSwFlux(
+        ::Type{FT};
+        nswf::FN3 = (t) -> FT(NaN),
+    ) where {FT, FN3}
+        args = nswf
+        return new{FT, FN3}(args)
+    end
+end
+
+"""
+    compute_net_sw_flux(
+        nswf_model::PrescribedSwFluxAndAlbedo{FT},
+        t::Real,
+    ) where {FT}
+
+Computes the net shortwave flux as a function of time.
+"""
+function compute_net_sw_flux(
+    nswf_model::PrescribedSwFluxAndAlbedo{FT},
+    t::Real,
+) where {FT}
+    net_sw_flux = (1 - nswf_model.α) * nswf_model.swf(t)
```

Comment:
if alpha can be a function of t (as it is now), then it should be evaluated here (t)

Reply:
FT(1)

Reply:
fixed alpha(t), why the FT(1) ?

Reply:
actually i think it's ok! the 1 will be promoted from an int to a float:
```
julia> typeof(1)
Int64
```
```
julia> typeof(1-Float32(1))
Float32
```
vs
```
julia> typeof(Float32(1)-Float32(1))
Float32
```

Reply:
this is good to know!